src/main/java/com/aaspaasdoctor/service/CalendarServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.aaspaasdoctor.service;
+
+import java.util.List;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.aaspaasdoctor.entity.Calendar;
+import com.aaspaasdoctor.repository.CalendarRepository;
+
+@Transactional
+@Service
+public class CalendarServiceImpl {
+
+	@Autowired
+	private CalendarRepository calendarRepository;
+
+	public Calendar addCalendarEvent(Calendar calendar) {
+		return calendarRepository.save(calendar);
+	}
+
+	public List<Calendar> getCalendarEventByPatientId(Integer pId) {
+		return calendarRepository.findByPId(pId);
+	}
+
+	public List<Calendar> getCalendarByDoctorId(Integer dId) {
+		return calendarRepository.findByDId(dId);
+	}
+
+	public Calendar updateCalendar(Calendar calendar) {
+		if (calendar != null && calendar.getCalendarId() != null) {
```

Comment:
Remove calendar != null && calendar.getCalendarId() != null condition as this is not User driven option ...when we will call update Calender its our Rest call in which CalenderId is not given by User so this condition never produce